rust/generate_message/src/lib.rs

```diff
@@ -1096,6 +1098,11 @@ use specs::gen_add_specs;
 mod error;
 pub use error::{Error, Result};
 
+lazy_static! {
+    pub(crate) static ref HOT_DB_PATH: PathBuf =
+        PathBuf::from(std::env::var("HOT_DB_PATH").unwrap_or_else(|_| HOT_DB_NAME.to_string()));
+}
```

Comment:
Would be cool to change these hardcoded paths to configuratble ones, so if this is starting to get changed I would propose just an `honest` cli flag instead of this env that would have a default value. That way it would have a `--help` doc and would be more usable by anyone

Reply:
So that would look like you add flags to top-level parsing structure


https://github.com/paritytech/parity-signer/blob/7ebe0d3eb45bc390ea2da7d715fbbcf2b7422224/rust/generate_message/src/parser.rs#L16-L20

that would replace these constant values.

Reply:
That was my initial approach as well. 

However, `generate_message` is used not only as a cli, but as a library as well.  I could add a `--hot-db-path` flag with a default value, but that would only work for cli consumers. Whoever is using `generate_message` as a library (e.g. `signer` crate or `metadata-portal`) would need to pass the value explicitly. 

Crates within `parity-signer` would need to have `constants` as a dependency just to import `HOT_DB_NAME` from there. 

For external users, it's even more confusing, because some commands (like `make` or `sign`) don't require a database, but users would be forced to pass `hot-db-path` argument anyway.

Example of `generate_message` usage:
https://github.com/paritytech/metadata-portal/blob/master/cli/src/updater/generate.rs#L28

Reply:
yeah well probably i was wrong about command-wide args and they can only be subcommand-specific flags for those subcommands that use them. And then the API that exists for use-cases like the portal could have an `Option`-al path value or something. In any case I am not a big fan of this

```rust
pub const HOT_DB_NAME: &str = "../database/database_hot";
```

We can do so much better.

Reply:
That makes sense. Fixed